web/client/components/TOC/fragments/MarkIcon.jsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { getWellKnownNameImageFromSymbolizer } from '../../../utils/VectorStyleUtils';
+
+function MarkIcon({ symbolizer }) {
+    const [iconData, setIconData] = useState(null);
+    useEffect(() => {
+        if (!symbolizer) {
+            return;
+        }
+        async function getIconData() {
+            const data = await getWellKnownNameImageFromSymbolizer(symbolizer);
+            setIconData(data);
+        }
+        getIconData();
```

Comment:
please use the promise instead o async/await, we currently did not setup babel to support them and in general we are not using them inside react components.
I would suggest to do something like this to apply icon data only if the component is still mounted

```js
const [iconData, setIconData] = useState(null);
const isMounted = useRef();

useEffect(() => {
    isMounted.current = true;
    return () => {
        isMounted.current = false;
    };
}, []);

useEffect(() => {
    getWellKnownNameImageFromSymbolizer(symbolizer)
        .then((data) => {
            if (isMounted.current) {
                setIconData(data);
            }
        });
}, [symbolizer]);
```